external-adapter/src/errors.ts

```diff
@@ -1,12 +1,19 @@
-class AdapterError extends Error {
+export class AdapterError extends Error {
+  jobRunID: string
+  status: string
+  statusCode: number
+  name: string
+  message: string
+  cause: any
+
   constructor({
     jobRunID = '1',
     status = 'errored',
     statusCode = 500,
     name = 'AdapterError',
     message = 'An error occurred.',
     cause,
-  }) {
+  }: Partial<AdapterError>) {
```

Comment:
Why would we need to type a class constructor as a `Partial` of that same class?

Reply:
All of the properties of the class are getting set here in the constructor, but are not actually required params (they have the default). The use of `Partial` here makes all of those params optional.